Sources/NIOCore/AsyncAwaitSupport.swift

```diff
@@ -240,8 +240,25 @@ extension ChannelPipeline {
     }
 }
 
+/// An error that is thrown when the number of bytes in an AsyncSequence exceeds the limit.
+///
+/// When collecting the bytes from an AsyncSequence, there is a limit up to where the content
+/// exceeds a certain threshold beyond which the content isn't matching an expected reasonable
+/// size to be processed. This error is generally thrown when it is discovered that there are more
+/// more bytes in a sequence than what was specified as the maximum. It could be that this upTo
+/// limit should be increased, or that the sequence has unexpected content in it.
 public struct NIOTooManyBytesError: Error, Hashable {
```

Comment:
Can you add a custom `Equatable` and `Hashable` implementation that ignores `maxBytes`? `==` can just return true and `hash(into:)` can just hash a constant e.g. `0`. Users are usually not interested in the specific `maxBytes` value but rather just that it is a `NIOTooManyBytesError` error.